homeassistant/components/device_tracker/geofency.py

```diff
@@ -0,0 +1,130 @@
+"""
+Support for the Geofency platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.geofency/
+"""
+import asyncio
+from functools import partial
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import PLATFORM_SCHEMA
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import (
+    ATTR_LATITUDE, ATTR_LONGITUDE, HTTP_UNPROCESSABLE_ENTITY, STATE_NOT_HOME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+BEACON_DEV_PREFIX = 'beacon'
+CONF_MOBILE_BEACONS = 'mobile_beacons'
+
+LOCATION_ENTRY = '1'
+LOCATION_EXIT = '0'
+
+URL = '/api/geofency'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_MOBILE_BEACONS): vol.All(
+        cv.ensure_list, [cv.string]),
+})
+
+
+def setup_scanner(hass, config, see, discovery_info=None):
+    """Set up an endpoint for the Geofency application."""
+    mobile_beacons = config.get(CONF_MOBILE_BEACONS) or []
+
+    hass.http.register_view(GeofencyView(see, mobile_beacons))
+
+    return True
+
+
+class GeofencyView(HomeAssistantView):
+    """View to handle Geofency requests."""
+
+    url = URL
+    name = 'api:geofency'
+
+    def __init__(self, see, mobile_beacons):
+        """Initialize Geofency url endpoints."""
+        self.see = see
+        self.mobile_beacons = [slugify(beacon) for beacon in mobile_beacons]
+
+    @asyncio.coroutine
+    def post(self, request):
+        """Geofency message received."""
+        data = yield from request.post()
+        return (yield from self._handle(request.app['hass'], data))
+
+    @asyncio.coroutine
+    def _handle(self, hass, data):
```

Comment:
Can you merge that into post function. A coroutine that wait for other coroutine slow done the system. So we need look that we do that only on points where a real reason exists.